swayfmt/src/comments.rs

```diff
@@ -0,0 +1,66 @@
+use std::fmt::Write;
+use sway_ast::token::Comment;
+use sway_types::Spanned;
+
+use crate::{formatter::FormattedCode, Formatter, FormatterError};
+
+// Given a start, an end and a CommentMap, return references to all comments
+// contained within a given start and end of a span, in an exclusive range.
+pub fn get_comments_between(start: usize, end: usize, formatter: &mut Formatter) -> Vec<&Comment> {
+    let mut comments = vec![];
+    let iter = formatter.comment_map.clone().into_keys();
```

Comment:
This clones the entire `comment_map` just to iterate over the keys. Instead, you could use the `keys()` iterator to iterate over the keys by reference, e.g. `formatter.comment_map.keys()`.

Reply:
Updated this to `comments_between()` - creating an iterator and returning it instead as you suggested below.